src/EditorFeatures/CSharpTest/ConvertCast/ConvertDirectCastToTryCastTests.cs

```diff
@@ -370,5 +371,68 @@ public static void Main()
                 CodeActionValidationMode = CodeActionValidationMode.SemanticStructure,
             }.RunAsync();
         }
+
+        [Theory]
+        [CombinatorialData]
+        public async Task ConvertFromExplicitToAs_NullableReferenceType(bool nullable)
+        {
+            var initialMarkup = $@"
+#nullable {(nullable ? "enable" : "disable")}
+
+class Program
+{{
+    public static void Main()
+    {{
+        var x = ([||]string?)null;
+    }}
+}}";
+            var expectedMarkup = $@"
+#nullable {(nullable ? "enable" : "disable")}
+
+class Program
+{{
+    public static void Main()
+    {{
+        var x = null as string;
+    }}
+}}";
+            await new VerifyCS.Test
+            {
+                TestCode = initialMarkup,
+                FixedCode = expectedMarkup,
+                CompilerDiagnostics = CompilerDiagnostics.None, // Suppress compiler warning about nullable string in non-nullable context
+                CodeActionValidationMode = CodeActionValidationMode.Full,
+            }.RunAsync();
+        }
```

Comment:
this is really complex to understand.  having two tests with the separate options/diagnostics would be cleaerer for me.